src/com/android/volley/Request.java

```diff
@@ -69,10 +69,12 @@
      * Supported request methods.
      */
     public enum ReturnStrategy {
-        DOUBLE, // You'll get called back into twice, once for cached result and once for network delivery
         NETWORK_IF_NO_CACHE, // You'll get the network response only if the cache misses
         CACHE_IF_NETWORK_FAILS, // You'll get the cached response only if the network errors / fails
-        NETWORK_ONLY // Skip the cache
+        NETWORK_ONLY, // Skip the cache
+        // END LEGACY RETURN STRATEGIES
+        DOUBLE, // You'll get called back into up to twice, once for cached result and once for network delivery
+        SINGLE // guarantees that response will be called only once
```

Comment:
depending on the cache settings, it sort of behaves has a hybrid of NETWORK_IF_NO_CACHE, CACHE_IF_NETWORK_FAILS and NETWORK_ONLY. I added it as a new type so I can support legacy types while adding divergent behavior for the SINGLE return type. Like for returning cache on error we can optimize by not storing the cache response, but by sending it back through the cache queue after network completes.